tests/contrib/django/test_django.py

```diff
@@ -1460,3 +1461,27 @@ def test_template_view_patching():
     # Patch via `as_view()`
     TemplateView.as_view()
     assert "dispatch" not in vars(TemplateView)
+
+
+def test_helper_get_request_uri():
```

Comment:
Is this the case that reproduces the issue?

Reply:
the bug is it `HTTP_HOST` is not bytes but the request path is? or vice versa?

Reply:
Added more test combinations for mixing the argument types to cover the 8 different combinations.